### ISSUE TITLE
Let cargo-generate read a config file with defaults.

### ISSUE THREAD

**Issue by najamelan** [cargo-generate/cargo-generate#210]
@ashleygwilliams I wonder how you feel about the possibility to set defaults, eg. if one often uses the same template, it would be nice to be able to omit the `--git` command line parameter. Also I would like to be able to turn off renaming a project with underscore to hyphen, saving another manual `mv` command. I imagine that other configuration options will come up once the infra is available.

Would you accept a PR for this? I would propose the user dir, something like `~/.config/cargo-generate.toml` and an equivalent on Windows.

ps: Let's be complete in saying that it can be worked around by creating a shell function to wrap cargo-generate, so it's not the end of the world.

**Comment by EverlastingBugstopper** [comment]
I'm not sure about the default git template functionality - that seems like something more suited to a shell alias than an explicit pattern.

As for turning off the renaming, I believe you should be able to do that with the `--force` flag already. This is not documented though - if you try it and it works, it may be worth adding a documentation PR.

**Comment by sassman** [comment]
@najamelan I was thinking about exactly the same. To give a bit more context, I imagined that the config file might contain something like:

```toml
[shortcuts]
cli = "https://github.com/sassman/cli-template-rs.git"
lib = "https://github.com/sassman/lib-template-rs.git"
```

and then I can use it this way:

```sh
cargo generate -s cli -name my-new-cli
# or the long version 
cargo generate --shortcut cli -name my-new-cli
```

**Comment by SirWindfield** [comment]
@sassman I really like that idea! Is there interest in getting something like this merged into `cargo-generate`?

**Comment by bilelmoussaoui** [comment]
I have a similar use case here https://gitlab.gnome.org/bilelmoussaoui/gtk-rust-template/-/issues/7 where I would like to provide a template for gtk3 & gtk4 without having to duplicate a bunch of files so I guess something like this would work somehow?

**Comment by taurr** [comment]
I might have the start for this - see #256

**Comment by sassman** [comment]
Hi @taurr,

your [feature/favorites](https://github.com/cargo-generate/cargo-generate/compare/master...taurr:feature-favorites) is even better than the proposal above. 

Do you want to add a few more tests and contribute with a PR? I would certainly appreciate it.

**Comment by taurr** [comment]
Hi @sassman 

I will add a few more tests tests, then make the PR - that was my intention from the get go, just didn't know how my proposal would be "judged" (first time Github contributer!) 😄

**Comment by sassman** [comment]
Any contribution that is helpful and reasonable is very welcome, and this seems to be useful to not only you. 

BTW congrats to your first time contribution on GitHub.

**Comment by sassman** [comment]
this has been implemented in [v0.6.0](https://github.com/cargo-generate/cargo-generate/releases/tag/v0.6.0)
